### ISSUE TITLE
Pytorch with ROCm on GFX1035?

### ISSUE THREAD

**Issue by Jonezia** [ROCm/ROCm#2048]
Hi,

I'm attempting to get Pytorch to work with ROCm on GFX1035 (AMD Ryzen 7 PRO 6850U with Radeon Graphics). I know GFX1035 is technically not supported, but it shares an instruction set with GFX1030 and others have had success building for GFX1031 and GFX1032 by setting HSA_OVERRIDE_GFX_VERSION=10.3.0.

I have ROCm setup, with rocminfo correctly displaying my GPU. Running "torch.cuda.is_available()" returns true. I am able to run sample neural networks on the CPU perfectly fine, and I have tested training on MNIST as well as inference on Resnet50. However, if I try and run the same code on the GPU it produces bizarre and incorrect results like producing labels indices that are negative and negative probabilities etc. I am able to run simple pytorch programs like sending two matrices to the gpu and multiplying them works correctly. However, with this setup even a simple neural network with one linear layer doesn't work.

Current setup:
Ubuntu 22.04.1 with kernel 5.15.0-43 generic
Python 3.9
ROCm 5.4.2 
Pytorch for ROCm 5.4.2 (bare metal)
I then run programs with HSA_OVERRIDE_GFX_VERSION=10.3.0
This is following someone's setup for GFX1032 that worked.

I have also tried Ubuntu 22.04.2 with kernel 5.17 OEM and Python3.10. This also produced wrong results on the Resnet50 too (but different), but I was even able to run a simple neural network of 2 linear layers, ReLU, and Sigmoid with this setup.

I was wondering if anybody has had any success with getting Pytorch to work with ROCm on GFX1035 and can share their setup or any help with this. I hope there is a setup that works if I can find it. 

Thanks for any help

**Comment by FNsi** [comment]
Check the data type? 
I think Fp16 are not working for gfx1035.

**Comment by Jonezia** [comment]
The data type is float32

**Comment by FNsi** [comment]
> The data type is float32

Normal models using fp16, and in many cases I cannot use fp16 in PyTorch apps, like realesr/stable defusion etc, but with --fp32 or --no-half.

**Comment by hongxiayang** [comment]
@Jonezia You might have PCIe atomics support issue. Can you run this test in your pytorch env:

```
python -c "import torch; x = torch.ones(1,2).cuda(); print(x); print('sum {}'.format(torch.sum(x)))"
```
Expected output is 2. If not, let me know what you see and please you check this https://github.com/pytorch/pytorch/issues/103973 for reference to understand this problem.

**Comment by ppanchad-amd** [comment]
@Jonezia Can you please test with latest ROCm 6.1.1? If resolved, please close ticket. Thanks!

**Comment by harkgill-amd** [comment]
Hi @Jonezia, I will close the ticket for now. If you still see this issue, please give the suggestion by @hongxiayang a try and reopen the issue. Thanks!
